unleash-yggdrasil/src/strategy_parsing.rs

```diff
@@ -418,7 +426,10 @@ fn get_hostname() -> CompileResult<String> {
 }
 
 #[cfg(not(feature = "hostname"))]
-fn get_hostname() -> CompileResult<String> {
+fn get_hostname(context: &Context) -> CompileResult<String> {
+    if let Some(hostname) = &context.runtime_hostname {
```

Comment:
The other piece of magic. If we have a hostname specified on the enriched context, we should use it